src/test/ui/consts/const_unsafe_unreachable_ub.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(const_fn)]
```

Comment:
If you make this test `#[warn(const_err)]` you'll get an additional error on the `BAR` use site, which you can then mark with `//~ ERROR erroneous constant`

Reply:
I tried to add `#[warn(const_err)]` on top of the BAR constant but I still don't get an error in the use site. Here's my diff and the stderr difference:
```diff
diff --git a/src/test/ui/consts/const_unsafe_unreachable_ub.rs b/src/test/ui/consts/const_unsafe_unreachable_ub.rs
index 2e4dfd1522b..1ca7a9e146d 100644
--- a/src/test/ui/consts/const_unsafe_unreachable_ub.rs
+++ b/src/test/ui/consts/const_unsafe_unreachable_ub.rs
@@ -1,3 +1,4 @@
+// check-pass
 #![feature(const_fn)]
 #![feature(const_unreachable_unchecked)]
 
@@ -8,6 +9,7 @@ const unsafe fn foo(x: bool) -> bool {
     }
 }
 
+#[warn(const_err)]
 const BAR: bool = unsafe { foo(false) };
 
 fn main() {
diff --git a/src/test/ui/consts/const_unsafe_unreachable_ub.stderr b/src/test/ui/consts/const_unsafe_unreachable_ub.stderr
index 0a7aa4e0021..340494fcf7b 100644
--- a/src/test/ui/consts/const_unsafe_unreachable_ub.stderr
+++ b/src/test/ui/consts/const_unsafe_unreachable_ub.stderr
@@ -1,4 +1,4 @@
-error: any use of this value will cause an error
+warning: any use of this value will cause an error
   --> $SRC_DIR/libcore/hint.rs:LL:COL
    |
 LL |     unsafe { intrinsics::unreachable() }
@@ -6,15 +6,19 @@ LL |     unsafe { intrinsics::unreachable() }
    |              |
    |              entering unreachable code
    |              inside `std::hint::unreachable_unchecked` at $SRC_DIR/libcore/hint.rs:LL:COL
-   |              inside `foo` at $DIR/const_unsafe_unreachable_ub.rs:7:18
-   |              inside `BAR` at $DIR/const_unsafe_unreachable_ub.rs:11:28
+   |              inside `foo` at $DIR/const_unsafe_unreachable_ub.rs:8:18
+   |              inside `BAR` at $DIR/const_unsafe_unreachable_ub.rs:13:28
    | 
-  ::: $DIR/const_unsafe_unreachable_ub.rs:11:1
+  ::: $DIR/const_unsafe_unreachable_ub.rs:13:1
    |
 LL | const BAR: bool = unsafe { foo(false) };
    | ----------------------------------------
    |
-   = note: `#[deny(const_err)]` on by default
+note: the lint level is defined here
+  --> $DIR/const_unsafe_unreachable_ub.rs:12:8
+   |
+LL | #[warn(const_err)]
+   |        ^^^^^^^^^
 
-error: aborting due to previous error
+warning: 1 warning emitted
 

```
Am I missing something?

Reply:
@oli-obk Thanks for the recommendation! I figured out and updated the code. Could you take a look again?
(Added it as a new commit to make it easier to see new changes, but I can squash with the previous commit later if you prefer it that way.)

Reply:
yea, that build-fail thing is really problematic around const prop and similar. I gotta figure something out for that.

The PR lgtm now, thanks!